app/views/datasets/_contact.html.erb

```diff
@@ -1,9 +1,11 @@
 <%
-  if contact_email_is_email?(dataset)
-    mail = mail_to(contact_email_for(dataset), nil, {'data-ga-event': 'contact', 'data-ga-publisher': @dataset.organisation.name, class: 'govuk_link'})
-  else
-    mail = link_to(contact_email_for(dataset), contact_email_for(dataset), {'data-ga-event': 'contact', 'data-ga-publisher': @dataset.organisation.name, class: 'govuk_link'})
-  end
+  default_name = "Contact #{@dataset.organisation.title} regarding this dataset"
+  link_attrs = {'data-ga-event': 'contact', 'data-ga-publisher': @dataset.organisation.name, 'class': 'govuk_link'}
+
+  email_name = contact_name_for(dataset).present? ? contact_name_for(dataset) : default_name
+  foi_name = foi_name_for(dataset).present? ? foi_name_for(dataset) : default_name
```

Comment:
```suggestion
  foi_name = foi_name_for(dataset).presence || default_name
```

I believe you can use the `presence` method here to simplify the code: https://apidock.com/rails/Object/presence